packages/@aws-cdk/aws-elasticloadbalancingv2/lib/shared/base-target-group.ts

```diff
@@ -245,6 +245,10 @@ export abstract class TargetGroupBase extends CoreConstruct implements ITargetGr
       this.setAttribute('deregistration_delay.timeout_seconds', baseProps.deregistrationDelay.toSeconds().toString());
     }
 
+    if (!cdk.Token.isUnresolved(baseProps.targetGroupName) && baseProps.targetGroupName !== undefined && (baseProps.targetGroupName.length > 32 || !/^[0-9a-z]+[0-9a-z-]*[0-9a-z]+$/i.test(baseProps.targetGroupName))) {
```

Comment:
Can you extract this logic into a separate function `validateTargetGroupName`? Same for `loadBalancerName`.

Reply:
I think better, I will move it to the new `validate` method coming from the new construct API (I say "new" here, but I mean new to me 😄)